### PR TITLE
Change error message and expand test coverage [Ready for Merging]

### LINKED ISSUE #214: ParserUtil Integer Overflow
[From AB3](https://github.com/nus-cs2103-AY2021S2/forum/issues/288)

### LINKED ISSUE #237: Improve test coverage for FindModuleCommand
Line coverage is around 50%, lower than that of other command classes

### PR DESCRIPTION
**Pull request by yhtMinceraft1010X**
For 0 and signed integer index inputs:
Now, it shows that index is not non-zero unsigned integer, rather than a generic error message.

### CODE REVIEW

**Review by AiwassPrime — COMMENTED** [review]
LGTM

File: docs/UserGuide.md [hunk 1]
```diff
@@ -378,6 +382,7 @@ Format: `find KEYWORD`
 
 * Searches through ModuleBook3.5 for tasks whose names contain `KEYWORD`.
 * You may key in multiple keywords. ModuleBook3.5 will list down all tasks that contains any of the given keyword.
+* Once another command is executed, even if it is not `list` command, the original unfiltered task list is restored for viewing.
```
Comment (QY-H00) [inline docs/UserGuide.md:391]: Nice explaination for this feature.

Comment (QY-H00) [inline docs/UserGuide.md:391]: Nice explaination for this feature.

File: src/main/java/seedu/module/commons/util/StringUtil.java [hunk 2]
```diff
@@ -49,20 +48,35 @@ public static String getDetails(Throwable t) {
...
         requireNonNull(s);
-
         try {
-            int value = Integer.parseInt(s);
-            return value > 0 && !s.startsWith("+"); // "+1" is successfully parsed by Integer#parseInt(String)
+            Integer.parseInt(s);
         } catch (NumberFormatException nfe) {
             return false;
         }
+        return true;
```
Comment (QY-H00) [inline src/main/java/seedu/module/commons/util/StringUtil.java:63]: Nice improvement on code quality.

File: src/test/java/seedu/module/commons/core/GuiSettingsTest.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
+package seedu.module.commons.core;
```
Comment (QY-H00) [inline src/test/java/seedu/module/commons/core/GuiSettingsTest.java:1]: Nice test for GUI settings.

File: src/test/java/seedu/module/commons/util/StringUtilTest.java [hunk 4]
```diff
@@ -13,33 +13,63 @@
...
+
+        // EP: positive unsigned numbers
+        assertTrue(StringUtil.isInteger("1")); // Boundary value
+        assertTrue(StringUtil.isInteger("10"));
+    }
+
+    @Test
+    public void isNonZeroUnsignedInteger() {
+
+        // EP: invalid inputs. Generate assertion error
```
Comment (QY-H00) [inline src/test/java/seedu/module/commons/util/StringUtilTest.java:52]: Is the reason here the method will directly detect that such string is not integer?

File: src/test/java/seedu/module/logic/commands/CommandTestUtil.java [hunk 5]
```diff
@@ -69,7 +69,7 @@
     public static final String INVALID_MODULE_DESC = " " + PREFIX_MODULE + "CS!2040"; // ! not allowed in modules
     public static final String INVALID_DESCRIPTION_DESC = " " + PREFIX_DESCRIPTION; // descriptions should not be empty
     public static final String INVALID_WORKLOAD_DESC = " " + PREFIX_WORKLOAD + "4"; // workload should be in range 1-3
-    public static final String INVALID_TAG_DESC = " " + PREFIX_TAG + "hubby*"; // '*' not allowed in tags
+    public static final String INVALID_TAG_DESC = " " + PREFIX_TAG + "quiz*"; // '*' not allowed in tags
```
Comment (QY-H00) [inline src/test/java/seedu/module/logic/commands/CommandTestUtil.java:72]: Nice modification.

File: src/test/java/seedu/module/logic/commands/SortCommandTest.java [hunk 6]
```diff
@@ -121,6 +121,8 @@ public void execute_sortedTaskByNumberOfTags_showsSortedList() {
 
     @Test
     public void equals() {
+        assertEquals(tagSortCommand, tagSortCommand);
```
Comment (QY-H00) [inline src/test/java/seedu/module/logic/commands/SortCommandTest.java:124]: Nice identical checking.

**Review by QY-H00 — COMMENTED** [review]
Overall LGTM!

File: src/test/java/seedu/module/commons/util/StringUtilTest.java (hunk 4 above)
Comment (yhtMinceraft1010X) [reply src/test/java/seedu/module/commons/util/StringUtilTest.java:52]: The string should have been processed by isInteger() first so a non-integer would not be passed into isNonZeroUnsignedInteger()
